test/repository/hbase/HBaseRestRepository_Delete_WiremockSpec.scala

```diff
@@ -0,0 +1,207 @@
+  ) {
+    val targetUrl = s"/${config.namespace}:$Table/$RowKey/$ColumnFamily:$ColumnName/?check=delete"
+  }
+
+  override protected def withFixture(test: OneArgTest): Outcome = {
+    val config = HBaseRestRepositoryConfig(
+      BaseUrl(protocol = "http", host = "localhost", port = wireMockPort, prefix = None),
+      "namespace", "username", "password", timeout = 1000L
+    )
+    val auth = Authorization(config.username, config.password)
+    val responseReaderMaker = stub[HBaseResponseReaderMaker]
+
+    WsTestClient.withClient { wsClient =>
+      withFixture(test.toNoArgTest(
+        FixtureParam(config, auth, new HBaseRestRepository(config, wsClient, responseReaderMaker), responseReaderMaker)
+      ))
+    }(new Port(wireMockPort))
+  }
+
+  private def stubTargetEntityIsFound(
+    config: HBaseRestRepositoryConfig,
+    authorization: Authorization,
+    responseReaderMaker: HBaseResponseReaderMaker,
+    withFields: Map[String, String] = DeleteField
+  ): Unit = {
+    val bodyParseResult = JsSuccess(Seq(Row(rowKey = "UnusedRowKey", withFields)))
+    stubTargetEntityRetrieves(config, authorization, responseReaderMaker, bodyParseResult)
+  }
+
+  private def stubTargetEntityIsNotFound(
+    config: HBaseRestRepositoryConfig,
+    authorization: Authorization,
+    responseReaderMaker: HBaseResponseReaderMaker
+  ): Unit =
+    stubTargetEntityRetrieves(config, authorization, responseReaderMaker, JsSuccess(Seq.empty))
+
+  private def stubTargetEntityRetrieves(
+    config: HBaseRestRepositoryConfig,
+    authorization: Authorization,
+    responseReaderMaker: HBaseResponseReaderMaker,
+    bodyParseResult: JsResult[Seq[Row]]
+  ): Unit = {
+    val DummyJsonResponseStr = """{"some":"json"}"""
+    val readsRows = stub[Reads[Seq[Row]]]
+
+    stubHBaseFor(getHBaseJson(s"/${config.namespace}:$Table/$RowKey/$ColumnFamily", authorization).willReturn(
+      anOkResponse().withBody(DummyJsonResponseStr)
+    ))
+    (responseReaderMaker.forColumnFamily _).when(ColumnFamily).returns(readsRows)
+    (readsRows.reads _).when(Json.parse(DummyJsonResponseStr)).returns(bodyParseResult)
+    () // explicitly return unit to avoid warning about disregarded return value
+  }
+
+  "A HBase REST Repository" - {
+    "successfully applies a delete" - {
+      "when the column exists and its value has not been modified by another user" in { fixture =>
+        stubTargetEntityIsFound(fixture.config, fixture.auth, fixture.responseReaderMaker)
+        stubHBaseFor(putHBaseJson(fixture.targetUrl, fixture.auth).withRequestBody(equalToJson(CheckAndDeleteRequestBody)).willReturn(
+          anOkResponse()
+        ))
+
+        whenReady(fixture.repository.deleteField(Table, RowKey, checkField = (QualifiedColumn, CurrentValue), columnName = QualifiedColumn)) { result =>
+          result shouldBe EditApplied
+        }
+      }
+
+      /*
+       * In this case we assume that either a previous attempt or someone else got their first.
```

Comment:
In this case we assume that **either a previous attempt** ... to edit the record has been made?

Reply:
Update the comment in an attempt to clarify.